src/index.js

```diff
@@ -1,30 +1,27 @@
 'use strict'
 
-const core = require('datastore-core')
-const MountStore = core.MountDatastore
-const ShardingStore = core.ShardingDatastore
-
-const Key = require('interface-datastore').Key
-const LevelStore = require('datastore-level')
 const waterfall = require('async/waterfall')
 const series = require('async/series')
 const parallel = require('async/parallel')
-const Multiaddr = require('multiaddr')
-const Buffer = require('safe-buffer').Buffer
+const each = require('async/each')
 const assert = require('assert')
 const path = require('path')
 const debug = require('debug')
 
+const backends = require('./backends')
 const version = require('./version')
 const config = require('./config')
+const apiAddr = require('./api-addr')
 const blockstore = require('./blockstore')
 const defaultOptions = require('./default-options')
 
 const log = debug('repo')
 
-const apiFile = new Key('api')
-const flatfsDirectory = 'blocks'
-const levelDirectory = 'datastore'
+const lockers = {
+  memory: require('./lock-memory'),
+  fs: require('./lock')
```

Comment:
Lock can't be a simple datastore as it uses os specific system calls on the filesystem. Though the memory lock could use something else as backend if needed. It might be good to allow passing in ones own lock module if one wants to instead of using one of the specified